rmf_site_editor/src/widgets/workspace.rs

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2024 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+use crate::widgets::{FileMenu, MenuDisabled, MenuEvent, MenuItem, TextMenuItem};
+use crate::{AppState, CreateNewWorkspace, WorkspaceLoader, WorkspaceSaver};
+use bevy::prelude::*;
+
+#[derive(Default)]
+pub struct WorkspaceMenuPlugin {}
+
+impl Plugin for WorkspaceMenuPlugin {
+    fn build(&self, app: &mut App) {
+        app.init_resource::<WorkspaceMenu>().add_systems(
+            Update,
+            handle_workspace_menu_events.run_if(AppState::in_displaying_mode()),
+        );
+    }
+}
```

Comment:
This is a new widget for workspace creation, now the top menu bar is fully modular and doesn't have any "prespawned" buttons.
The reason this is needed is again modularity and code duplication. The top menu bar buttons forced the widget to rely on internal workspace creation / file loading / file saving events, this meant that users that wanted custom workflows for these functions would need to duplicate the whole top menu bar.
Moving them to a classic child system means that users just need to add their children to the file menu instead.

This is necessary because these kind of file operations will most likely have to be customized in different apps. For example a file loading might need custom extension filters, a different loaded file handler etc.